### PR TITLE
Customizable Rocker Amplitude

### PR DESCRIPTION
**Pull request by Coronia**
Help TwinkleStar to finish his pull request, with proper docs this time
#1080

The rocker amplitude of warheads with `Rocker=yes` used to be determined by `Damage` value of the weapon. You can now override it with another fixed value or add a mutiplier to it.

In `rulesmd.ini`:
```ini
[SOMEWARHEAD]       ; WarheadType
Rocker.AmplitudeOverride=         ; integer
Rocker.AmplitudeMultiplier=1.0    ; double
```

### CODE REVIEW

File: src/Ext/TechnoType/Body.h [hunk 1]
```diff
@@ -191,6 +191,8 @@ class TechnoTypeExt
 		Valueable<TechnoTypeClass*> Convert_HumanToComputer;
 		Valueable<TechnoTypeClass*> Convert_ComputerToHuman;
 
+		Valueable<AnimTypeClass*> Wake;
+
```
Comment (Metadorius) [inline src/Ext/TechnoType/Body.h:195]: Wouldn't it be better to use Nullable here too?

**Review by Metadorius — APPROVED** [review]
No complaints about code style, probably would be better to use Nullable here tho.

**Review by chaserli — CHANGES_REQUESTED** [review]
Apparently you missed the wakes for squids and crashed units.
rebase/force push the branch plz
Do not abuse sanity check when it's pointless

**Comment by Metadorius** [comment]
> rebase/force push the branch plz

I already merged the latest dev and fixed the build.

File: src/Ext/TechnoType/Body.cpp [hunk 2]
```diff
@@ -92,6 +92,23 @@ void TechnoTypeExt::ExtData::ParseBurstFLHs(INI_EX& exArtINI, const char* pArtSe
 	}
 }
 
+AnimTypeClass* TechnoTypeExt::GetWakeAnim(LocomotionClass* pLoco)
+{
+	const auto pThis = pLoco->LinkedTo;
+	if (!pThis)
+		return nullptr;
+
+	const auto pExt = TechnoTypeExt::ExtMap.Find(pThis->GetTechnoType());
```
Comment (FS-21) [inline src/Ext/TechnoType/Body.cpp:101]: should be called pTypeExt instead of pExt

**Comment by Coronia** [comment]
TwinkleStar has completed the customized wake settings for Squid parasite and sinking as well

In `rulesmd.ini`:
```ini
[SOMETECHNO]         ; TechnoType
Wake.Grapple=        ; Anim (played when Techno being parasited on the water), default to [General] Wake
Wake.Sinking=        ; Anim (played when Techno sinking), default to [General] Wake
```

**Comment by Metadorius** [comment]
Took the liberty of splitting off mergeable changes and merging them with correct credit.

As for the `Rocker`, I think there are some improvements to be done.

```ini
[SOMEWH]                        ; WarheadType
Rocker.AmplitudeMultiplier=1.0  ; float
Rocker.FixedAmplitude=          ; integer
```

The first tag would multiply the regular damage if the user wants it to scale with damage, the second one would just override the damage with a fixed one. Also better naming.

**Comment by Metadorius** [comment]
GH screw off, I didn't want to close this

**Comment by Coronia** [comment]
> Took the liberty of splitting off mergeable changes and merging them with correct credit.
> 
> As for the `Rocker`, I think there are some improvements to be done.
> 
> ```ini
> [SOMEWH]                        ; WarheadType
> Rocker.AmplitudeMultiplier=1.0  ; float
> Rocker.FixedAmplitude=          ; integer
> ```
> 
> The first tag would multiply the regular damage if the user wants it to scale with damage, the second one would just override the damage with a fixed one. Also better naming.

added the suggested change. The second tag is named as Rocker.AmplitudeOverride

**Pull request merged**